openhtf/__init__.py

```diff
@@ -161,14 +162,27 @@ def plug_type_map(self):
     return plug_type_map
 
 
-def SetupFramework():
-  """Sets up various bits of the framework. Only needs to be called once."""
-  try:
-    sys.argv = FLAGS(sys.argv)  # parse flags
-  except gflags.FlagsError, e:  # pylint: disable=invalid-name
-    print '%s\nUsage: %s ARGS\n%s' % (e, sys.argv[0], FLAGS)
-    sys.exit(1)
+def CreateArgParser():
+  """Creates an argparse.ArgumentParser for parsing command line flags."""
+  parser = argparse.ArgumentParser('OpenHTF-based testing')
+  # openhtf.io.user_input
+  # Option 1:
+  parser.add_argument('--prompt_timeout_s', type=int,
+                      help='User prompt timeout in seconds.')
+  # Option 2:
+  parser = argparse.ArgumentParser('OpenHTF-based testing', parents=[
+      user_input.DEFAULT_TIMEOUT_S_PARSER])
+
+
+  return parser
 
+
+@functions.RunOnce
+def SetupFramework(arguments=None):
+  """Sets up various bits of the framework. Only needs to be called once."""
+  parser = arguments or CreateArgParser().parse_args()
+  # Option 3:
```

Comment:
After some discussion this morning, @grybmadsci and I recommend option 3, with the added caveat of taking the flag's default from a pseudo-constant defined at the top of `user_input.py`. 

The concern being that it's weird to have a flag default here, and a default-default in the `user_input` module. It's hard in that case to understand which "default" is being respected, and you might change it one place without knowing about its use in the other. Taking the default from the `user_input` module and using it at flag declaration here alleviates the confusion of multiple defaults, and also preserves the one-way relationship of `user_input` not needing to make any assumptions about how it's being used.